File: Official Windows Platform Sample/Windows 8 app samples/[VB]-Windows 8 app samples/VB/Windows 8 app samples/XAML FlipView control sample (Windows 8)/VB/DataModel/SampleDataSource.vb
Imports System
Imports System.Linq
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.ComponentModel
Imports System.Runtime.CompilerServices
Imports Windows.ApplicationModel.Resources.Core
Imports Windows.Foundation
Imports Windows.Foundation.Collections
Imports Windows.UI.Xaml.Data
Imports Windows.UI.Xaml.Media
Imports Windows.UI.Xaml.Media.Imaging

' The data model defined by this file serves as a representative example of a strongly-typed
' model that supports notification when members are added, removed, or modified.  The property
' names chosen coincide with data bindings in the standard item templates.
'
' Applications may use this model as a starting point and build on it, or discard it entirely and
' replace it with something appropriate to their needs.

Namespace Data
    ''' <summary>
    ''' Base class for <see cref="SampleDataItem"/> and <see cref="SampleDataGroup"/> that
    ''' defines properties common to both.
    ''' </summary>
    <Windows.Foundation.Metadata.WebHostHidden> _
    Public MustInherit Class SampleDataCommon
        Inherits SDKTemplate.Common.BindableBase

        Private Shared _baseUri As New Uri("ms-appx:///")

        Public Sub New(title As String, type As String, picture As String)
            Me._title = title
            Me._type = type
            Me._picture = picture
        End Sub

        Private _title As String = String.Empty
        Public Property Title() As String
            Get
                Return Me._title
            End Get
            Set(value As String)
                Me.SetProperty(Me._title, value)
            End Set
        End Property

        Private _type As String = String.Empty
        Public Property Type() As String
            Get
                Return Me._type
            End Get
            Set(value As String)
                Me.SetProperty(Me._type, value)
            End Set
        End Property

        Private _image As Uri = Nothing
        Private _picture As String = Nothing
        Public Property Image() As Uri
            Get
                Return New Uri(SampleDataCommon._baseUri, Me._picture)
            End Get

            Set(value As Uri)
                Me._picture = Nothing
                Me.SetProperty(Me._image, value)
            End Set
        End Property
    End Class

    ''' <summary>
    ''' Generic item data model.
    ''' </summary>
    Public Class SampleDataItem
        Inherits SampleDataCommon
        Public Sub New(title As String, type As String, picture As String)
            MyBase.New(title, type, picture)
        End Sub

    End Class

    ''' <summary>
    ''' Creates a collection of groups and items with hard-coded content.
    ''' </summary>
    Public NotInheritable Class SampleDataSource
        Private _items As New ObservableCollection(Of Object)()
        Public ReadOnly Property Items() As ObservableCollection(Of Object)
            Get
                Return Me._items
            End Get
        End Property

        Public Sub New()
            Items.Add(New SampleDataItem("Cliff", "item", "Assets/Cliff.jpg"))
            Items.Add(New SampleDataItem("Grapes", "item", "Assets/Grapes.jpg"))
            Items.Add(New SampleDataItem("Rainier", "item", "Assets/Rainier.jpg"))
            Items.Add(New SampleDataItem("Sunset", "item", "Assets/Sunset.jpg"))
            Items.Add(New SampleDataItem("Valley", "item", "Assets/Valley.jpg"))
        End Sub
    End Class
End Namespace

File: Official Windows Platform Sample/Windows Phone 8.1 samples/111541-Windows Phone 8.1 samples/XAML FlipView control sample/C#/Shared/DataModel/SampleDataSource.cs
using System;
using System.Linq;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel;
using System.Runtime.CompilerServices;
using Windows.ApplicationModel.Resources.Core;
using Windows.Foundation;
using Windows.Foundation.Collections;
using Windows.UI.Xaml.Data;
using Windows.UI.Xaml.Media;
using Windows.UI.Xaml.Media.Imaging;

// The data model defined by this file serves as a representative example of a strongly-typed
// model that supports notification when members are added, removed, or modified.  The property
// names chosen coincide with data bindings in the standard item templates.
//
// Applications may use this model as a starting point and build on it, or discard it entirely and
// replace it with something appropriate to their needs.

namespace Controls_FlipView.Data
{
    /// <summary>
    /// Base class for <see cref="SampleDataItem"/> and <see cref="SampleDataGroup"/> that
    /// defines properties common to both.
    /// </summary>
    [Windows.Foundation.Metadata.WebHostHidden]
    public abstract class SampleDataCommon : SDKTemplate.Common.BindableBase
    {
        private static Uri _baseUri = new Uri("ms-appx:///");

        public SampleDataCommon(String title, String type, String picture)
        {
            this._title = title;
            this._type = type;
            this._picture = picture;
        }

        private string _title = string.Empty;
        public string Title
        {
            get { return this._title; }
            set { this.SetProperty(ref this._title, value); }
        }

        private string _type = string.Empty;
        public string Type
        {
            get { return this._type; }
            set { this.SetProperty(ref this._type, value); }
        }

        private Uri _image = null;
        private String _picture = null;
        public Uri Image
        {
            get
            {
                return new Uri(SampleDataCommon._baseUri, this._picture);
            }

            set
            {
                this._picture = null;
                this.SetProperty(ref this._image, value);
            }
        }
    }

    /// <summary>
    /// Generic item data model.
    /// </summary>
    public class SampleDataItem : SampleDataCommon
    {
        public SampleDataItem(String title, String type, String picture)
            : base(title, type, picture)
        {
        }

    }

    /// <summary>
    /// Creates a collection of groups and items with hard-coded content.
    /// </summary>
    public sealed class SampleDataSource
    {
        private ObservableCollection<object> _items = new ObservableCollection<object>();
        public ObservableCollection<object> Items
        {
            get { return this._items; }
        }

        public SampleDataSource()
        {
            Items.Add(new SampleDataItem("Cliff",
                    "item",
                    "Assets/Cliff.jpg"
                    ));
            Items.Add(new SampleDataItem("Grapes",
                    "item",
                    "Assets/Grapes.jpg"
                    ));
            Items.Add(new SampleDataItem("Rainier",
                    "item",
                    "Assets/Rainier.jpg"
                    ));
            Items.Add(new SampleDataItem("Sunset",
                    "item",
                    "Assets/Sunset.jpg"
                    ));
            Items.Add(new SampleDataItem("Valley",
                    "item",
                    "Assets/Valley.jpg"
                    ));
        }
    }
}
